java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -1656,6 +1656,175 @@ public void zlexcount(BaseClient client) {
         assertTrue(executionException.getCause() instanceof RequestException);
     }
 
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void zrangestore_by_index(BaseClient client) {
+        String key = "{testKey}:" + UUID.randomUUID();
+        String destination = "{testKey}:" + UUID.randomUUID();
+        String source = "{testKey}:" + UUID.randomUUID();
+        Map<String, Double> membersScores = Map.of("one", 1.0, "two", 2.0, "three", 3.0);
+        assertEquals(3, client.zadd(source, membersScores).get());
+
+        // Full range.
+        assertEquals(3, client.zrangestore(destination, source, new RangeByIndex(0, -1)).get());
+        assertEquals(
+                Map.of("one", 1.0, "two", 2.0, "three", 3.0),
+                client.zrangeWithScores(destination, new RangeByIndex(0, -1)).get());
+
+        // Range from rank 0 to 1. In descending order of scores.
+        assertEquals(2, client.zrangestore(destination, source, new RangeByIndex(0, 1), true).get());
+        assertEquals(
+                Map.of("three", 3.0, "two", 2.0),
+                client.zrangeWithScores(destination, new RangeByIndex(0, -1)).get());
+
+        // Incorrect range as start > stop.
+        assertEquals(0, client.zrangestore(destination, source, new RangeByIndex(3, 1)).get());
+        assertEquals(Map.of(), client.zrangeWithScores(destination, new RangeByIndex(0, -1)).get());
+
```

Comment:
Do we want any of the following tests here or do you think they are unnecessary?
- stop value greater than the length of the sorted set
- limit arg

Reply:
There is no limit arg for RangeByIndex. No need to test the first one it just gives you everything.